gateway/src/proxy.rs

```diff
@@ -97,10 +100,11 @@ async fn proxy(
     let target_ip = if let Some(ip) = { state.project_cache.get(project_name.as_str()) } {
         ip
     } else {
-        let ip = state
+        let (proj, was_stopped) = state
             .gateway
             .find_or_start_project(&project_name, state.task_sender.clone())
-            .await?
+            .await?;
+        let ip = proj
             .state
             .target_ip()?
             .ok_or_else(|| Error::from_kind(ErrorKind::ProjectNotReady))?;
```

Comment:
nit: I guess this can be another source of 502? It is probably less problematic than the deployer getting back up and starting the runtime.

Reply:
Perhaps, but unrelated to this change